### PR TITLE
#165020084 create and view an article

### PR DESCRIPTION
**Pull request by dusmel**
#### What does this PR do?
- This PR adds 2 components to handle the creation and view of an article, plus a toast to handle server messages and a progress bar to add on good user experience.

#### Description of Task to be completed?
As a user you should be able to create articles with a simplistic and flexible editor, you can add as many images as you want, plus the whole page is responsive to mobile and large screens. After creating an article you will be redirected to the display page which comes with all the article details and the author information.

#### How should this be manually tested?
after login go to `articles/new` add the title and the body of the article then submit

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#165020084](https://www.pivotaltracker.com/story/show/165020084)

#### Screenshots (if appropriate)

# 1 Creat article
image

# 2 upload an image

ezgif com-video-to-gif

# 3 Mobile View
image

# 4 Toast for server messages
image

# 5 View one article
image

#### Questions:

N/A

### CODE REVIEW

**Review by giseleiradu — APPROVED** [review]
Good job, Dusmel.
Something I would recommend is to fix those CI violations. Otherwise, it is good.

File: src/Containers/Article/Create/createArticle.jsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      ...currentUser,
+      profile: {
+        username,
+      },
+    };
+    const { editorState, loadingBarProgress } = this.state;
+    return (
+      <div className="article-create">
+        <Navbar {...this.props} currentUser={currentUser} />
+        <LoadingBar progress={loadingBarProgress} height={3} color="red" />
```
Comment (Karlmusingo) [inline src/Containers/Article/Create/createArticle.jsx:171]: Can you please change this color and take our primary color?

File: src/reducers/system.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  noInternet: false,
+};
+
+/**
+ *
+ * @param {object} state
+ * @param {object} action
+ * @returns {object} new state
+ */
+const login = (state = initialState, { type, payload }) => {
```
Comment (Karlmusingo) [inline src/reducers/system.js]: Can you please change the name of this reducer into something else more relevant here?

File: src/Containers/Article/Create/createArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+let images = [];
+let spinner = false;
+
+/**
+ *
+ * @param {object} editorState
+ * @returns {method} render
+ */
+export class CustomImageSideButton extends ImageSideButton {
```
Comment (Karlmusingo) [inline src/Containers/Article/Create/createArticle.jsx:27]: Can you find a way to modularize this file by moving this class into its own file?

**Review by Karlmusingo — APPROVED** [review]

**Pull request merged**